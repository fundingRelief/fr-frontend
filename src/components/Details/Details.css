.Details {
  margin-top: 5rem;
  font-family: 'Comfortaa', cursive;

  a {
    text-decoration: none;
  }

  a:link {
    color: rgba(40, 167, 118, 0.6);
  }

  a:visited {
    color: rgba(40, 167, 118, 0.6);
  }

  a:hover {
    color: rgb(50, 150, 207);
  }

  a:active {
    color: rgba(40, 167, 118, 0.6);
  }

  .Info {
    text-align: center;
  }
  .Description {
    text-align: center;
  }
}
