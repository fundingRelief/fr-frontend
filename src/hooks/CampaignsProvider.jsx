// import { useState, useEffect } from 'react';
import React, { createContext, useState, useContext, useEffect } from 'react';
import PropTypes from 'prop-types';
import { fetchCampaigns } from '../services/fundingReliefAPI';

const CampaignsContext = createContext();

export const CampaignsProvider = ({ children }) => {
  const [campaigns, setCampaigns] = useState([]);
  const [loading, setLoading] = useState(false);
  const [lastPage, setLastPage] = useState(null);

  useEffect(() => {
    setLoading(true);
    fetchCampaigns()
      .then(fetchedCampaigns => {
        setCampaigns(fetchedCampaigns);
        setLoading(false);
      });
  }, []);

  return (
    <CampaignsContext.Provider
      value={{
        campaigns,
        loading,
        setLoading,
        lastPage,
        setLastPage
      }}
    >
      {children}
    </CampaignsContext.Provider>
  );
};

CampaignsProvider.propTypes = {
  children: PropTypes.node,
};

export const useCampaign = () => {
  const { campaigns } = useContext(CampaignsContext);
  return campaigns;
};

export const useLoading = () => {
  const { loading } = useContext(CampaignsContext);
  return loading;
};

export const useSetLoading = () => {
  const { setLoading } = useContext(CampaignsContext);
  return setLoading;
};

export const useLocalCampaigns = () => {
  const { localCampaigns } = useContext(CampaignsContext);
  return localCampaigns;
};

export const useLastPage = () => {
  const { lastPage } = useContext(CampaignsContext);
  return lastPage;
};

export const useSetLastPage = () => {
  const { setLastPage } = useContext(CampaignsContext);
  return setLastPage;
};


// localCampaigns

// export const useGetCampaigns = fetchFunction => {
//   const [campaigns, setCampaigns] = useState([]);

//   const [loading, setLoading] = useState(false);



//   // console.log(campaigns);

//   // if(campaigns.length === 0) {
//   //   useEffect(() => {
//   //     setLoading(true);
//   //     fetchFunction()
//   //       .then(fetchedCampaigns => {
//   //         setCampaigns(fetchedCampaigns);
//   //         setLoading(false);
//   //       });
//   //   }, []);
//   // }

//   useEffect(() => {
//     setLoading(true);
//     fetchFunction()
//       .then(fetchedCampaigns => {
//         setCampaigns(fetchedCampaigns);
//         setLoading(false);
//       });
//   }, []);

//   // useEffect(() => {
//   //   setLoading(true);
//   //   fetchFunction()
//   //     .then(fetchedCampaigns => {
//   //       setCampaigns(fetchedCampaigns);
//   //       setLoading(false);
//   //     });
//   // }, []);

//   return { campaigns, setCampaigns, loading, setLoading };
// };


