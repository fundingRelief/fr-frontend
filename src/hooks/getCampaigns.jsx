
import { useState, useEffect } from 'react';

export const useGetChallenges = fetchFunction => {
  const [campaigns, setCampaigns] = useState([]);

  const [loading, setLoading] = useState(false);

  useEffect(() => {
    setLoading(true);
    fetchFunction()
      .then(fetchedCampaigns => {
        setCampaigns(fetchedCampaigns);
        setLoading(false);
      });
  }, []);

  // useEffect(() => {
  //   setSelectedChallenges(category === 'allCategories' ? challenges : challenges.filter(challenge => challenge.category === category));
  // }, [category]);

  return { campaigns, setCampaigns, loading, setLoading };
};
