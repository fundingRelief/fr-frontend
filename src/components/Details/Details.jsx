import React from 'react';
import PropTypes from 'prop-types';
import { Icon, Image, Container, Header } from 'semantic-ui-react';
import { Link } from 'react-router-dom';
import styles from './Details.css';
import { useLastPage } from '../../hooks/CampaignsProvider';

const Details = ({
  campaign_name,
  current_amount,
  goal,
  percentage_raised,
  location,
  img_url,
  link_url,
  description,
  last_donation,
}) => {
  const numberWithCommas = (x) => {
    return x.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',');
  };
  const lastPage = useLastPage();
  return (
    <>
      <Container text className={styles.Details}>
        <Image
          centered
          bordered rounded size="large"
          src={img_url} alt="Fund being described."
        />
        <Header textAlign={'center'}>{campaign_name}</Header>
        <p className={styles.Whatever}>
          {description}
        </p>
        {location}
        {last_donation}
        ${numberWithCommas(current_amount)} out of ${numberWithCommas(goal)}{' '} goal
        {percentage_raised}% fulfilled
        
        <Icon name="user" href={link_url} target="_blank" rel="noopener noreferrer">
            Donate Now GoFundMe
        </Icon>
        <br />
        <Link to={lastPage}>back</Link>

      </Container>
    </>
  );
};

Details.propTypes = {
  id: PropTypes.number,
  campaign_name: PropTypes.string.isRequired,
  current_amount: PropTypes.number.isRequired,
  goal: PropTypes.number.isRequired,
  percentage_raised: PropTypes.number.isRequired,
  location: PropTypes.string.isRequired,
  img_url: PropTypes.string.isRequired,
  link_url: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  last_donation: PropTypes.string.isRequired,
  cause: PropTypes.string.isRequired,
};

export default Details;
