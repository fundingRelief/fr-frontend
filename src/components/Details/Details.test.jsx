import React from 'react';
import { shallow } from 'enzyme';
import Details from './Details';
import { CampaignsProvider } from '../../hooks/CampaignsProvider';

describe('Details component', () => {
  it('renders Details', () => {
    const wrapper = shallow(
      <CampaignsProvider>
        <Details
          campaign_name="Our parents lost their home to the Almeda fire"
          current_amount={944}
          goal={5000}
          percentage_raised={18}
          location="PORTLAND, OR"
          img_url="https://d2g8igdw686xgo.cloudfront.net/51544330_1601348243982129_r.jpeg"
          link_url="https://gf.me/u/y2z2h8"
          description="On September 8th our parents and many other families lost their home in Phoenix, Oregon. Our parents lost their place to relax and escape to after work. We as a family lost the place we knew we could walk right in without knocking and raid the fridge. The place where my boys would go to and run around in the backyard and play with Paco our family dog. The place where my brother would come back home to every summer after teaching in Oklahoma as teacher for Teach for America. The place where my sister would come home to when she would get homesick or just need a break from being a student at Portland State University. We decided to start this GoFundMe for our parents who have sacrificed for us their entire lives, they always put our needs ahead of theirs even to this day even after losing their home. We hope that we can do our part for them now. Thank you."
          last_donation="Last donation 2h ago"
          cause="https://www.gofundme.com/c/act/oregon-fires"
        />
      </CampaignsProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
