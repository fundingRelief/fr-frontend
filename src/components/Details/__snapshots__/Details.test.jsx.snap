// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details component renders Details 1`] = `
<Fragment>
  <h3>
    Our parents lost their home to the Almeda fire
  </h3>
  <h4>
    944
     / 
    5000
  </h4>
  <h4>
    18.8
    % filled with 
    18
     donors
  </h4>
  <img
    src="https://d2g8igdw686xgo.cloudfront.net/51544330_1601348243982129_r.jpeg"
  />
  <a
    href="https://gf.me/u/y2z2h8"
    rel="noopener noreferrer"
    target="_blank"
  >
    link
  </a>
  <h4>
    PORTLAND, OR
  </h4>
  <p>
    On September 8th our parents and many other families lost their home in Phoenix, Oregon. Our parents lost their place to relax and escape to after work. We as a family lost the place we knew we could walk right in without knocking and raid the fridge. The place where my boys would go to and run around in the backyard and play with Paco our family dog. The place where my brother would come back home to every summer after teaching in Oklahoma as teacher for Teach for America. The place where my sister would come home to when she would get homesick or just need a break from being a student at Portland State University. We decided to start this GoFundMe for our parents who have sacrificed for us their entire lives, they always put our needs ahead of theirs even to this day even after losing their home. We hope that we can do our part for them now. Thank you.
  </p>
</Fragment>
`;
